File: lambda/ws-disconnect.ts
import { APIGatewayProxyWebsocketEventV2 } from "aws-lambda";
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import {
  DynamoDBDocumentClient,
  DeleteCommand,
} from "@aws-sdk/lib-dynamodb";

const ddb = DynamoDBDocumentClient.from(new DynamoDBClient({}));
const TABLE = process.env.CONNECTIONS_TABLE!;

export const handler = async (event: APIGatewayProxyWebsocketEventV2) => {
  const connectionId = event.requestContext.connectionId!;

  await ddb.send(
    new DeleteCommand({
      TableName: TABLE,
      Key: { connectionId },
    })
  );

  return { statusCode: 200, body: "Disconnected." };
};

File: lambda/ws_disconnect.py
import os
import boto3

dynamodb = boto3.resource("dynamodb")
table = dynamodb.Table(os.environ.get("TABLE_NAME", "ConnectionTable"))

def handler(event, context):
    connection_id = event["requestContext"]["connectionId"]

    # delete the connection
    table.delete_item(Key={"ConnectionId": connection_id})

    return {
        "statusCode": 200,
        "body": "Disconnected."
    }
